tests/blackbox/scopes/test_bb_scopes_updates.py

```diff
@@ -453,3 +452,96 @@ def test_scopes_linkage_update(setup_scopez_server_action):
     assert l_r.status_code == 403
     assert l_r.text == 'This is rules custom response\n'
 
+# ------------------------------------------------------------------------------
+# test /update_bots endpoint
+# ------------------------------------------------------------------------------
+def test_update_bots_endpoint(setup_scopez_server_action):
+    # ------------------------------------------------------
+    # test for recieving a bot challenge with attack vector
+    # ------------------------------------------------------
+    l_url = G_TEST_HOST + '/update_bots'
+
+    l_file_path = os.path.dirname(os.path.abspath(__file__))
+    l_test_file = os.path.realpath(os.path.join(l_file_path,
+                                                '../../data/waf/conf/bots/0052-wHyMHxV7.bots.json'))
+    l_test_payload = ''
+
+    # check setup
+    assert os.path.exists(l_test_file), 'test file not found!'
+
+    # slurp test file
+    with open(l_test_file) as l_tf:
+        l_test_payload = l_tf.read()
+
+    # check setup
+    assert l_test_payload, 'payload is empty!'
+
+    l_json_payload = json.loads(l_test_payload)
+
+    # check positive test
+    l_result = requests.post(l_url, timeout=3, json=l_json_payload)
+    assert l_result.status_code == 200
+    assert l_result.json()['status'] == 'success'
+
```

Comment:
you actually dont need this POST to test if the challenge works , because configs gets loaded in the startup and this POST is techinically not updating anything , so you should be able to get challenge directly without this POST. 1. check challenge works 2. then change somethings in the config and do POST etc.. rest of the logic looks fine

Reply:
oh right, i meant to remove that, thanks for reminder